contract/models/res_partner.py

```diff
@@ -7,30 +7,63 @@
 class ResPartner(models.Model):
     _inherit = 'res.partner'
 
-    contract_count = fields.Integer(
-        string='Contracts',
+    sale_contract_count = fields.Integer(
+        string='Sale Contracts',
+        compute='_compute_contract_count',
+    )
+    purchase_contract_count = fields.Integer(
+        string='Purchase Contracts',
         compute='_compute_contract_count',
     )
 
     def _compute_contract_count(self):
-        Contract = self.env['account.analytic.account']
+        contract_model = self.env['account.analytic.account']
         today = fields.Date.today()
         for partner in self:
-            partner.contract_count = Contract.search_count([
+            partner.sale_contract_count = contract_model.search_count([
                 ('recurring_invoices', '=', True),
                 ('partner_id', 'child_of', partner.ids),
                 '|',
                 ('date_end', '=', False),
                 ('date_end', '>=', today),
+                ('contract_type', '=', 'sale'),
             ])
+            partner.purchase_contract_count = contract_model.search_count([
+                ('recurring_invoices', '=', True),
+                ('partner_id', 'child_of', partner.ids),
+                '|',
+                ('date_end', '=', False),
+                ('date_end', '>=', today),
+                ('contract_type', '=', 'purchase'),
+            ])
+
+    def act_show_sale_contract(self):
```

Comment:
Please use a common method called from both methods that uses an argument with the XML-ID, but don't repeat the very similar code.